### ISSUE TITLE
bug: cards not formatting correctly in Teams when using TeamsQueueHandler

### ISSUE THREAD

**Issue by idexxbernvaug** [AnesFoufa/python-teams-logger#14]
Using logging dict configuration in django application

This configuration works:
```
'formatters': {
        'teamscard': {
            '()': Office365CardFormatter,
            'facts': ["name", "levelname", "lineno"],
        },
    },
'handlers': {
        'msteams': {
            'level': 'INFO',
            'class': 'teams_logger.TeamsHandler',
            'url': os.environ.get('TEAMS_LOGGING_URL', default=""),
            'formatter': 'teamscard',
        },
    },
```
the cards are displayed as normal in Teams:
![image](https://user-images.githubusercontent.com/81988675/122079053-0cb25100-cdcb-11eb-9939-d4e4b03bac2a.png)

However, I need a non-blocking implementation, so I changed to TeamsQueueHandler
```
'formatters': {
        'teamscard': {
            '()': Office365CardFormatter,
            'facts': ["name", "levelname", "lineno"],
        },
    },
'handlers': {
        'msteams': {
            'level': 'INFO',
            'class': 'teams_logger.TeamsQueueHandler',
            'url': os.environ.get('TEAMS_LOGGING_URL', default=""),
            'formatter': 'teamscard',
        },
    },
```
But this breaks the card formatting in Teams:
![image](https://user-images.githubusercontent.com/81988675/122079339-500cbf80-cdcb-11eb-968e-895dab8bd57d.png)

Is there a workaround that I have missed?

**Comment by AnesFoufa** [comment]
Excuse my late answer. I'll try to reproduce the issue and propose a fix.

**Comment by wvanhed** [comment]
The issue seems to be that the `self.formatter` which is - at some point- set on the TeamsQueueHandler, is not transferred to the `teams_handler` that is given to the `QueueListener`. 

I've tried addressing that with the following changes

    class TeamsQueueHandler(QueueHandler):
        def __init__(self, url, level=NOTSET):
            self._log_queue = queue.Queue(-1)
            super().__init__(self._log_queue)
            
            self._teams_handler = TeamsHandler(url, level).     # changed to self._teams_handler
            teams_log_listener = QueueListener(self._log_queue, self._teams_handler). # changed
            teams_log_listener.start()

        def setFormatter(self, fmt):          # override Handler baseclass method
            self._teams_handler.setFormatter(fmt)
            super().setFormatter(fmt)

But that didn't work out completely as expected. The cards are now shown with fields as expected, but the raw json content (`{@context...`) is inserted as subtitel in the card (instead of the log message). Couldn't really yet figure out why in the time I decided to spent on this.

**Comment by AnesFoufa** [comment]
I just published a release (0.4.1) wuth your suggested fixes
